docs/design-patterns.md

```diff
@@ -0,0 +1,152 @@
+# Design Patterns
+
+The main operating principle for `vdom` is:
```

Comment:
`vdom` (`virtual DOM`)

The non-frontend folks will thank you for the definition.

Reply:
Oh right! Thank you.

Reply:
Reflecting on this today I wish I called the library `layout` or something. I guess someone could build on top of `vdom` and abstract away the innards here.